tests/simulation_framework/test_cache_model.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+import tempfile
+from typing import Optional
+
+import pydantic as pd
+
+import flow360.component.simulation.units as u
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.framework.cache_model_base import CacheModelBase
+from flow360.component.simulation.unit_system import (
+    DensityType,
+    LengthType,
+    TemperatureType,
+)
+
+
+class TempThermalStateCache(Flow360BaseModel):
+    altitude: Optional[LengthType.Positive] = None
+    temperature_offset: Optional[TemperatureType] = None
+
+
+class TempThermalState(CacheModelBase):
+    temperature: TemperatureType.Positive = pd.Field(288.15 * u.K, frozen=True)
```

Comment:
Could you add a todo here? We can probably somehow give user warning that the give altitude (if any) will not be valid anymore if the altitude was used and user try to change these fields.